### ISSUE TITLE
MPO fine tune and infer

### ISSUE THREAD

**Issue by ruian1** [OpenGVLab/InternVL#1217]
I fine-tuned the 8B with mpo method, while the training logs look ok (except accuracy maybe? it bumps between 0 and 1 at the beginning and quickly stable at 1), the inference shows repeated symbols. I wanted to get some help to see if there are anything wrong from below so I know where to debug..

Follow-up, I do notice the model parameters become all 0s

1. mpo traning log
```{'loss': 1.6719, 'grad_norm': 9.85250186920166, 'learning_rate': 5.468308211179496e-10, 'rewards/chosen': 5.90625, 'rewards/rejected': -3.453125, 'rewards/accuracies': 1.0, 'rewards/margins': 9.375, 'logps/rejected': -123.0, 'logps/chosen': -160.0, 'logits/rejected': 0.0986328125, 'logits/chosen': 0.15625, 'nll_loss': 1.734375, 'epoch': 0.99}
 99%|█████████▊| 410/416 [2:54:16<02:32, 25.47s/it]dynamic ViT batch size: 26, images per sample: 13.0, dynamic token length: 12288
dynamic ViT batch size: 26, images per sample: 13.0, dynamic token length: 12288
[2025-10-22 19:18:24,259] [INFO] [logging.py:128:log_dist] [Rank 0] time (ms) | optimizer_step: 2439.20
[2025-10-22 19:18:24,262] [INFO] [logging.py:128:log_dist] [Rank 0] time (ms) | fwd_microstep: 4716.46 | bwd_microstep: 13375.23 | bwd_inner_microstep: 13370.97 | bwd_allreduce_microstep: 4.14 | step_microstep: 2603.30
[2025-10-22 19:18:24,263] [INFO] [logging.py:128:log_dist] [Rank 0] time (ms) | fwd: 4716.44 | bwd: 13375.23 | bwd_inner: 13370.96 | bwd_allreduce: 4.16 | step: 2603.31
{'loss': 1.3594, 'grad_norm': 7.992441177368164, 'learning_rate': 3.797647809745119e-10, 'rewards/chosen': 7.0, 'rewards/rejected': -2.75, 'rewards/accuracies': 1.0, 'rewards/margins': 9.75, 'logps/rejected': -146.0, 'logps/chosen': -103.0, 'logits/rejected': 0.00482177734375, 'logits/chosen': 0.10009765625, 'nll_loss': 1.2734375, 'epoch': 0.99}
 99%|█████████▉| 411/416 [2:54:41<02:07, 25.48s/it]dynamic ViT batch size: 26, images per sample: 13.0, dynamic token length: 12288
dynamic ViT batch size: 26, images per sample: 13.0, dynamic token length: 12288
[2025-10-22 19:18:48,255] [INFO] [logging.py:128:log_dist] [Rank 0] time (ms) | optimizer_step: 2429.32
[2025-10-22 19:18:48,257] [INFO] [logging.py:128:log_dist] [Rank 0] time (ms) | fwd_microstep: 4430.44 | bwd_microstep: 12493.44 | bwd_inner_microstep: 12490.39 | bwd_allreduce_microstep: 2.95 | step_microstep: 2587.87
[2025-10-22 19:18:48,258] [INFO] [logging.py:128:log_dist] [Rank 0] time (ms) | fwd: 4430.42 | bwd: 12493.44 | bwd_inner: 12490.37 | bwd_allreduce: 2.97 | step: 2587.87
{'loss': 1.4043, 'grad_norm': 9.3011474609375, 'learning_rate': 2.4306053790357217e-10, 'rewards/chosen': 9.875, 'rewards/rejected': -4.4375, 'rewards/accuracies': 1.0, 'rewards/margins': 14.3125, 'logps/rejected': -164.0, 'logps/chosen': -52.75, 'logits/rejected': 0.41015625, 'logits/chosen': 0.08544921875, 'nll_loss': 0.9765625, 'epoch': 0.99}
 99%|█████████▉| 412/416 [2:55:05<01:40, 25.04s/it]dynamic ViT batch size: 28, images per sample: 14.0, dynamic token length: 12288
dynamic ViT batch size: 28, images per sample: 14.0, dynamic token length: 12288
[2025-10-22 19:19:13,733] [INFO] [logging.py:128:log_dist] [Rank 0] time (ms) | optimizer_step: 2421.40
[2025-10-22 19:19:13,736] [INFO] [logging.py:128:log_dist] [Rank 0] time (ms) | fwd_microstep: 4725.63 | bwd_microstep: 13376.36 | bwd_inner_microstep: 13372.01 | bwd_allreduce_microstep: 4.22 | step_microstep: 2577.62
[2025-10-22 19:19:13,736] [INFO] [logging.py:128:log_dist] [Rank 0] time (ms) | fwd: 4725.61 | bwd: 13376.35 | bwd_inner: 13371.99 | bwd_allreduce: 4.25 | step: 2577.62
{'loss': 0.9917, 'grad_norm': 11.500261306762695, 'learning_rate': 1.3672639938777653e-10, 'rewards/chosen': 10.75, 'rewards/rejected': -4.90625, 'rewards/accuracies': 1.0, 'rewards/margins': 15.625, 'logps/rejected': -227.0, 'logps/chosen': -35.0, 'logits/rejected': 0.43359375, 'logits/chosen': 0.318359375, 'nll_loss': 0.65234375, 'epoch': 0.99}
 99%|█████████▉| 413/416 [2:55:31<01:15, 25.17s/it]dynamic ViT batch size: 18, images per sample: 9.0, dynamic token length: 12288
dynamic ViT batch size: 18, images per sample: 9.0, dynamic token length: 12288
[2025-10-22 19:19:39,330] [INFO] [logging.py:128:log_dist] [Rank 0] time (ms) | optimizer_step: 2422.22
[2025-10-22 19:19:39,333] [INFO] [logging.py:128:log_dist] [Rank 0] time (ms) | fwd_microstep: 4777.26 | bwd_microstep: 13450.99 | bwd_inner_microstep: 13444.13 | bwd_allreduce_microstep: 6.64 | step_microstep: 2583.34
[2025-10-22 19:19:39,334] [INFO] [logging.py:128:log_dist] [Rank 0] time (ms) | fwd: 4777.24 | bwd: 13450.99 | bwd_inner: 13444.12 | bwd_allreduce: 6.77 | step: 2583.35
```

2. model loading function, checkpoint_path points to the model
```
        model = InternVLChatModel.from_pretrained(
            checkpoint,
            low_cpu_mem_usage=False,
            torch_dtype=torch.bfloat16,
            device_map=None,
        ).to('cuda')
```

3. However the outputs are like below, before mpo fine-tuning the reslut makes sense
```
(((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((((&
```

4. One example of traing data
```
{"question": "<image><image> question", # ~50 tokens
  "chosen": sentence 1, # ~100-200 tokens
  "rejected": sentence 2, # ~100-200 tokens
  "answer_gt": sentence 1 # in our case answer_gt is same as chosen1
}
```

5. All zeros in model
```
    model = InternVLChatModel.from_pretrained(model_path, torch_dtype=torch.bfloat16)
    
    print("\n✓ Model loaded successfully")
    print(f"\nModel config:")
    print(f"  use_llm_lora: {model.config.use_llm_lora}")
    print(f"  use_backbone_lora: {model.config.use_backbone_lora}")
    
    # Check language model
    print(f"\nLanguage model type: {type(model.language_model)}")
    print(f"Language model has 'merge_and_unload': {hasattr(model.language_model, 'merge_and_unload')}")
    
    # Check vision model
    print(f"\nVision model type: {type(model.vision_model)}")
    print(f"Vision model has 'merge_and_unload': {hasattr(model.vision_model, 'merge_and_unload')}")
    
    # Sample a few weights
    print(f"\nLanguage model first layer weight sample:")
    first_param = next(model.language_model.parameters())
    print(f"  Shape: {first_param.shape}")
    print(f"  Mean: {first_param.mean().item():.6f}")
    print(f"  Std: {first_param.std().item():.6f}")
    print(f"  Min: {first_param.min().item():.6f}")
    print(f"  Max: {first_param.max().item():.6f}")
    
```

✓ Model loaded successfully

Model config:
  use_llm_lora: 16
  use_backbone_lora: 0

Language model type: <class 'peft.peft_model.PeftModelForCausalLM'>
Language model has 'merge_and_unload': True

Vision model type: <class 'internvl.model.internvl_chat.modeling_intern_vit.InternVisionModel'>
Vision model has 'merge_and_unload': False

Language model first layer weight sample:
  Shape: torch.Size([151674, 3584])
  Mean: 0.000000
  Std: 0.000000
  Min: 0.000000
  Max: 0.000000
  ❌ WARNING: All zeros!
```

```

**Comment by ruian1** [comment]
updates, found that the saved model after training is all zeros

**Comment by ruian1** [comment]
start checkpoint was not lora-merged
